spec/models/labware_creators/cardinal_pools_plate_spec.rb

```diff
@@ -60,6 +61,22 @@
       expect(Rails.application.config.cardinal_pooling_config[26]).to eq(2)
       expect(Rails.application.config.cardinal_pooling_config[20]).to eq(1)
     end
+
+    context 'when missing a sample manifest' do
+      it 'fails validation when all wells are missing a sample manifest' do
+        stub_v2_plate(plate, stub_search: false)
+        plate.wells.map { |well| well.aliquots.first.sample.sample_manifest = nil }
+        expect(subject).to_not be_valid
+        expect(subject.errors.messages[:source_plate]).to be_present
+      end
+
+      it 'fails validation when 1 well is missing a sample manifest' do
+        stub_v2_plate(plate, stub_search: false)
+        plate.wells[0].aliquots.first.sample.sample_manifest = nil
+        expect(subject).to_not be_valid
+        expect(subject.errors.messages[:source_plate]).to be_present
+      end
+    end
```

Comment:
It might be worth an extra test where it has a sample manifest, but that doesn't have the supplier filled out - if it's possible to create a manifest with no supplier? Because that would hit a different bit of your `if well.aliquots.first.sample.sample_manifest&.supplier_name.nil?`